rust/feature-flags/src/api/request_handler.rs

```diff
@@ -290,20 +290,22 @@ pub fn decode_request(
         .and_then(|v| v.to_str().ok())
         .unwrap_or("unknown");
 
-    if content_type.starts_with("application/json; encoding=base64")
-        && !matches!(query.compression, Some(Compression::Base64))
-    {
+    // Special case: if the Content-Type header explicitly specifies base64 encoding,
+    // this takes precedence over the query parameter for compression. In this case,
+    // the body is decoded as base64, and the query parameter is disregarded.
+    if content_type.starts_with("application/json; encoding=base64") {
```

Comment:
Shouldn't we parse the content type and then compare. For example, this is a valid content type: `Content-Type: application/json; charset=utf-8; encoding=base64`

Reply:
per slack: I'm just going to explicitly support the content types that we send from our SDKs.